MIGRATION.md

```diff
@@ -0,0 +1,188 @@
+- Ability to specify additional IAM Policies to attach to the node role
+- Ability to set whether or not the `AmazonEKS_CNI_Policy` is attached to the node role
+- Ability to provide your own IAM Role for the node group so you have complete control over its settings
+- Ability to specify node group placement details via `placement`
+- Ability to enable Nitro Enclaves on Nitro instances
+- Ability to configure Terrafrom create, update, and delete timeouts
+
+We also take advantage of improved AWS support for managed node upgrades. Now things like changing security groups or disk size no longer require a full replacement of the node group but instead are handled by EKS as rolling upgrades. This release includes support for the new `update_config` configuration that sets limits on how many nodes can be out of service during an upgrade.
+
+
+
+See the [README](https://github.com/cloudposse/terraform-aws-eks-node-group) for more details.
+
+## Breaking changes in v0.25.0
+
+Releases v0.11.0 through v0.20.0 of this module attempted to maintain compatiblity, so that no code changes were needed to upgrade and node groups would not likely be
+recreated on upgrade. Releases between v0.20.0 and v0.25.0 were never
+recommended for use because of compatibility issues. With the release
+of v0.25.0 we are making significant, breaking changes in order to bring
+this module up to current Cloud Posse standards. Code changes will likely
+be needed and node groups will likely need to be recreated. We strongly recommend
+enabling `create_before_destroy` if you have not already, as in general
+it provides a better upgrade path whenever an upgrade or change in confguration requires an node group to be replaced.
+
+### Terraform Version
+
+Terraform version 1.0 is out. Before that, there was Terraform version 0.15, then 0.14, then 0.13 and so on. The v0.25.0 release of this module allows you to try to use it with Terraform 0.13, but if it fails, we are not going to do anything about it. That version is old and has lots of known issues. There are hardly any breaking changes between Terraform 0.13 and 1.0, so please upgrade to the latest Terraform version before raising any issues about this module.
+
+### Behavoir changes
+
+- Previously, EBS volumes were left with the default value of `delete_on_termination`, which is `true` for EKS AMI root volumes. Now the default EBS volume has it set to `true` explicitly.
+- Previously, the Instance Metadata Service v1 (IMDSv1) was enabled by default, which is considered a security risk. Now it is disabled by default. Set `metadata_http_tokens_required` to `false` to leave IMDSv1 enabled.
+- Previously, a launch template was only generated and used if the settings required a launch template to set them. Now a launch template is alway generated (unless a launch template ID is provided) and used, and anything that can be set in the launch template is set there rather than in the node group configuration.
```

Comment:
```suggestion
- Previously, a launch template was only generated and used if the settings required a launch template to be created. Now a launch template is alway generated (unless a launch template ID is provided) and used, and anything that can be set in the launch template is set there rather than in the node group configuration.
```